src/server/json_family.cc

```diff
@@ -286,6 +287,36 @@ string ConvertToJsonPointer(string_view json_path) {
   return result;
 }
 
+void CountJsonFieldsImpl(const json& j, size_t& res) {
+  json_type type = j.type();
+  if (type == json_type::array_value) {
+    res += j.size();
+    for (const auto& item : j.array_range()) {
+      if (item.type() == json_type::array_value || item.type() == json_type::object_value) {
+        CountJsonFieldsImpl(item, res);
+      }
+    }
+
+  } else if (type == json_type::object_value) {
+    res += j.size();
+    for (const auto& item : j.object_range()) {
+      if (item.value().type() == json_type::array_value ||
+          item.value().type() == json_type::object_value) {
+        CountJsonFieldsImpl(item.value(), res);
+      }
+    }
+
+  } else {
+    res += 1;
+  }
+}
+
+size_t CountJsonFields(const json& j) {
+  size_t res = 0;
+  CountJsonFieldsImpl(j, res);
+  return res;
+}
```

Comment:
Just a side note, if you always return the additional size, you don't need a separate impl function

Reply:
According to your suggestion, I removed the separate implementation function, Thanks!